sdk/include/reactos/shellfolderutils.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
+ * PURPOSE:     Utility functions for IShellFolder implementations
+ * COPYRIGHT:   Copyright 2024 Whindmar Saksit <whindsaks@proton.me>
+ */
+
+#pragma once
+
+#ifdef __cplusplus
+
+template <BOOL LOGICALCMP = TRUE>
+static HRESULT ShellFolderImpl_CompareItemColumn(IShellFolder2 *psf, UINT column, PCUITEMID_CHILD pidl1, PCUITEMID_CHILD pidl2)
+{
+    SHELLDETAILS details1, details2;
+    LPWSTR str1, str2;
+    HRESULT hr;
+    if (SUCCEEDED(hr = psf->GetDetailsOf(pidl1, column, &details1)))
```

Comment:
Just a suggestion, could you [flatten](https://reactos.org/wiki/Coding_Style#Control_structures) this pyramid of doom, please?